src/platforms/stm32/gdb_if.c

```diff
@@ -118,8 +123,10 @@ unsigned char gdb_if_getchar_to(int timeout)
 
 	if (!(out_ptr < count_out)) do {
 		/* Detach if port closed */
-		if (!cdcacm_get_dtr())
-			return 0x04;
+			if (!cdcacm_get_dtr()) {
+				__WFI(); /* systick will wake up too!*/
+				return 0x04;
+			}
```

Comment:
One indentation level too many

Reply:
The line if () do { causes the strange ident.